go/src/staleFlagCleaner.go

```diff
@@ -0,0 +1,624 @@
+limitations under the License.
+*/
+package src
+
+import (
+	"encoding/json"
+	"go/token"
+	"io/ioutil"
+	"log"
+	"os"
+
+	/*
+		We have used dst package instead of official go ast package because ast package
+		does not keep positions of comment in place. So, we use dst package which keeps
+		the positions of comment in place. It is similiar to ast package and have same
+		functions as ast. So you can use documentation of ast to see the details of helper
+		functions.
+	*/
+	"github.com/dave/dst"
+	"github.com/dave/dst/dstutil"
+)
+
+// API : For the type of API
+type API int
+
+const (
+	isTreated API = iota
+	isControl
+	isTesting
+	isUnknown
+)
+
+// Methods : For json it is
+type Methods struct {
+	Methods []FlagAPI `json:"methodProperties"`
+}
+
+// FlagAPI : for getting the type from config file
+type FlagAPI struct {
+	MethodName string `json:"methodName"`
+	OfType     string `json:"flagType"`
+	FlagIndex  int    `json:"argumentIndex"`
+}
+
+// Value : This is made to not confuse with true and false in code
+type Value int
+
+const (
+	isTrue Value = iota
+	isFalse
+	// This will be returned when you don't have to find true or false
+	isBot
+)
+
+// Operator : To get the boolean operators
+type Operator int
+
+const (
+	and Operator = iota
+	or
+	not
+)
+
+// Some useful global variable
+var treated, control string = "treated", "control"
+var strTrue, strFalse string = "true", "false"
+
+// Implementing this as whole class
+type staleFlagCleaner struct {
+	configFile string
+	FlagAPIArr []FlagAPI
+	methods    Methods
+
+	/*
+		Here valueMap contains those which came from the stale function output.
+		It means that this will used for cleaning in Deep Clean Pass
+	*/
+	valueMap map[string]Value
+	functype map[string]FlagAPI
+
+	flagName  string
+	isTreated bool
+}
+
+func (sfc *staleFlagCleaner) init(configFile string, flagName string, isTreated bool) {
+	sfc.configFile = configFile
+	sfc.flagName = flagName
+	sfc.isTreated = isTreated
+	sfc.valueMap = make(map[string]Value)
+	sfc.functype = make(map[string]FlagAPI)
+	sfc.ParseJSON()
+}
+
+// Parse json
+func (sfc *staleFlagCleaner) ParseJSON() {
+
+	jsonFile, err := os.Open(sfc.configFile)
+
+	if err != nil {
+		log.Fatalf("failed opening file: %s", err)
+		os.Exit(-1)
+	}
+
+	byteValue, _ := ioutil.ReadAll(jsonFile)
+
+	// we unmarshal our byteArray which contains our
+	// jsonFile's content into 'methodProperties' which we defined above
+	json.Unmarshal(byteValue, &sfc.methods)
+	// store it in a hashmap
+	for _, element := range sfc.methods.Methods {
+		sfc.functype[element.MethodName] = element
+	}
+}
+
+// Gets the type of the flagAPI
+func (sfc *staleFlagCleaner) flagTypeAPI(callExpr *dst.CallExpr) API {
+	var apiName string
+	var flagIndex int // -1 if flag or group not here
+	flagIndex = -1
+
+	for ind, expr := range callExpr.Args {
+		switch arg := expr.(type) {
+		case *dst.Ident:
+			if arg.Name == sfc.flagName {
+				flagIndex = ind
+				break
+			}
+		}
+	}
```

Comment:
Do we need this? Wouldn't it be better to just check for the argument at `element.FlagIndex` below, right above line 152, to see if it matches  `sfc.flagName` and skip otherwise?

Reply:
Yeah, we don't need it. Dunno, why I didn't seen it.